metricbeat/module/vsphere/datastore/datastore_test.go

```diff
@@ -51,28 +52,29 @@ func TestFetchEventContents(t *testing.T) {
 	assert.EqualValues(t, "OTHER", event["fstype"])
 
 	// Values are based on the result 'df -k'.
-	fields := []string{"capacity.total.bytes", "capacity.free.bytes",
-		"capacity.used.bytes"}
+	fields := []string{
+		"capacity.total.bytes",
+		"capacity.free.bytes",
+		"status",
+		"host.count",
+		"vm.count",
+		"write.bytes",
+		"capacity.used.bytes",
+	}
 	for _, field := range fields {
 		value, err := event.GetValue(field)
 		if err != nil {
-			t.Error(err)
-		} else {
-			isNonNegativeInt64(t, field, value)
+			t.Error(field, err)
+			return
 		}
-	}
-}
-
-func isNonNegativeInt64(t testing.TB, field string, v interface{}) {
-	i, ok := v.(int64)
-	if !ok {
-		t.Errorf("%v: got %T, but expected int64", field, v)
-		return
-	}
+		if field == "status" {
+			assert.NotNil(t, value)
+		} else if field == "vm.count" || field == "host.count" {
+			assert.GreaterOrEqual(t, value, 0)
```

Comment:
Use `switch` statement instead
```suggestion
		switch field {
		case "status":
			assert.NotNil(t, value)
		case "vm.count", "host.count":
			assert.GreaterOrEqual(t, value, 0)
		default:
			assert.GreaterOrEqual(t, value, int64(0))
		}
```